src/components/todo-form/form.component.js

```diff
@@ -0,0 +1,180 @@
+    option.innerHTML = project.title;
+    projectListselect.append(option);
+    Form.saveDataToTheLocalStorage(`${project.title}Color`, { color: project.color });
+    Form.closeForm(projectFormContainer);
+    return project;
+  }
+
+  static getProjectFormInfo() {
+    const projectForm = document.getElementById('project-form');
+    const projectFormContainer = document.getElementById('project-form-container');
+    projectForm.addEventListener('click', (e) => {
+      if (e.target.id === 'btn-create-project') {
+        const project = Form.getProjectTitleAndColor(projectForm);
+        return Form.addNewProjectToForm(project);
+      }
+
+      if (e.target.id === 'cancel-project-button') {
+        Form.closeForm(projectFormContainer);
+        return false;
+      }
+      return false;
+    });
+  }
+
+  static lookExistenceProject(formData) {
+    const storageData = Form.getDataforTheLocalStorage('todoApp');
+    if (storageData) {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const storageElement of storageData) {
+        if (storageElement.name === formData.name) {
+          storageElement.todos.push(formData.todos[0]);
+          return storageData;
+        }
+      }
+    }
+    storageData.push(formData);
+    return storageData;
+  }
+
+  static saveData(form) {
+    const id = nanoid(10);
+    const project = form.elements.project.value;
+    const projectStorageData = window.localStorage.getItem(`${project}Color`);
+    const title = form.elements.title.value;
+    const date = form.elements.date.value;
+    const description = form.elements.description.value;
+    const priority = form.elements.priority.value;
+    if (projectStorageData === null) Form.saveDataToTheLocalStorage(`${project}Color`, { color: '#d01de0' });
+    const color = Form.getDataforTheLocalStorage(`${project}Color`);
+    const todo = new Todo(
+      id,
+      project,
+      title,
+      date,
+      description,
+      priority,
+    );
+    if (!Form.formValidation(todo)) return false;
+
+    const formData = { name: project, color: color.color, todos: [todo] };
+    const storageProject = Form.lookExistenceProject(formData);
+    Form.saveDataToTheLocalStorage('todoApp', storageProject);
+    return true;
+  }
+
+  static addOptionsFromStorage() {
+    const projectList = document.getElementById('project-list');
+    const storageProjects = Form.getDataforTheLocalStorage('todoApp');
+    if (storageProjects) {
+      storageProjects.forEach(project => {
+        if (project.name !== 'default') {
+          const opinionTag = document.createElement('option');
+          opinionTag.value = project.name;
+          opinionTag.id = project.name;
+          opinionTag.innerHTML = project.name;
+          projectList.append(opinionTag);
+        }
+      });
+    }
+  }
+
+  getFromData() {
+    Form.addOptionsFromStorage();
+    Form.getProjectFormInfo();
+    this.form.addEventListener('click', (e) => {
```

Comment:
- [ ] If I tried to add a todo without description, the page reloads but the error message briefly appears before reloading the page. To fix this issue I suggest that you add ` e.preventDefault();` in submitting the forms.